### PR TITLE
Feature/tensorflow tinyyolo

### PR DESCRIPTION
**Pull request by migueltaylor**
Postprocess no longer needs the `GstVideoInfo` of the model. Should we remove it on this branch?

### CODE REVIEW

File: ext/r2inference/gsttinyyolo.c [hunk 1]
```diff
@@ -41,26 +41,39 @@
...
-/* Box dim */
-#define BOX_DIM 4
+#define BOXES 5
+/* 
+ * Box dim 
+ * [0]: x (center)
+ * [0]: y (center)
+ * [2]: height
+ * [3]: width
+ * [4]: Objectness score
```
Comment (michaelgruner) [inline ext/r2inference/gsttinyyolo.c]: Objectness? Objetiveness?

File: ext/r2inference/gsttinyyolo.c [hunk 2]
```diff
@@ -41,26 +41,39 @@
...
+#define BOX_DIM 5
+/* Objectness threshold */
+#define OBJ_THRESH 0.08
 /* Probability threshold */
 #define PROB_THRESH 0.08
 /* Intersection over union threshold */
 #define IOU_THRESH 0.30
 
+const gfloat box_anchors[] =
+    { 1.08, 1.19, 3.42, 4.41, 6.63, 11.38, 9.42, 5.11, 16.62, 10.52 };
```
Comment (michaelgruner) [inline ext/r2inference/gsttinyyolo.c]: What are these hardcoded values?

File: ext/r2inference/gsttinyyolo.c [hunk 3]
```diff
@@ -226,68 +240,61 @@ gst_tinyyolo_finalize (GObject * object)
...
-    gint image_height)
+    gint image_height, gint box)
 {
-
   /* adjust the box center according to its cell and grid dim */
-  normalized_box->x += col;
-  normalized_box->y += row;
-  normalized_box->x /= GRID_H;
-  normalized_box->y /= GRID_W;
+  normalized_box->x = (col + sigmoid (normalized_box->x)) * 32.0;
```
Comment (michaelgruner) [inline ext/r2inference/gsttinyyolo.c]: What is this magic 32? can you use it through a const variable with an appropriate name?

File: ext/r2inference/gsttinyyolo.c [hunk 4]
```diff
@@ -226,68 +240,61 @@ gst_tinyyolo_finalize (GObject * object)
...
-  normalized_box->width *= normalized_box->width;
-  normalized_box->height *= normalized_box->height;
-
-  /* scale the boxes to the image size in pixels */
-  normalized_box->x *= image_width;
-  normalized_box->y *= image_height;
-  normalized_box->width *= image_width;
-  normalized_box->height *= image_height;
+  normalized_box->width =
+      pow (M_E, normalized_box->width) * box_anchors[2 * box] * 32.0;
```
Comment (michaelgruner) [inline ext/r2inference/gsttinyyolo.c]: Same as above

File: ext/r2inference/gsttinyyolo.c (hunk 1 above)
Comment (migueltaylor) [reply ext/r2inference/gsttinyyolo.c]: Objectness: The state of being an object

File: ext/r2inference/gsttinyyolo.c (hunk 2 above)
Comment (migueltaylor) [reply ext/r2inference/gsttinyyolo.c]: This implementation of TinyYOLO uses "box anchors" to force the model to detect boxes on different scales. It detects 5 boxes for each grid cell, each cell is 32x32 pixels, the first anchor [1.08,1.19] means that in box number one the biggest object that can be detected is 32*1.08 pixels wide and 32*1.19 pixels tall.

**Pull request merged**